File: webapp/src/types/frames.ts
export type Act = {
  id: string
  act: string
  actor: string
  action: string
  object: string
  recipient: string
  preconditions: {
    expression: string
    operand: boolean
  }
  create: string[]
  terminate: string[]
  sources: string[]
  explanation: string
}

export type Fact = {
  id: string
  fact: string
  function: string[]
  sources: string[]
  explanation: string
}

export type Duty = {
  id: string
  duty: string
  dutyHolder: string
  claimant: string
  terminatingAct: string[]
  creatingAct: string[]
  enforcingAct: string
  sources: string[]
}

export type Frames = {
  acts?: Act[]
  facts?: Fact[]
  duties?: Duty[]
}

export type SentenceAndFrames = {
  sentence: string
  frames: Frames
}

export type GameDetails = {
  game: string
  details: SentenceAndFrames[]
}

type HandleDeleteFrameType = (sentence: string, index: number) => void
type HandleAddFrameType = (sentence: string, frameType: string) => void

export type RuleDetailsProps = {
  sentence: string
  frames: Frames
  onDelete: HandleDeleteFrameType
  onFrameAdd: HandleAddFrameType
  onFrameEdit: any
}

File: nlp-api/frames.py
def create_empty_flint_format() -> dict:
    flint_format = {
        "sentence":"",
        "acts": [],
        "facts": [],
        "duties": []
    }
    return flint_format

def create_empty_act_frame() -> dict:
    act_frame = {
        "id":"",
        "act": "",
        "actor": "",
        "action": "",
        "object": "",
        "recipient": "",
        # "preconditions": {
        #     "expression": "LITERAL",
        #     "operand": True
        # },
        "preconditions": [],
        "create": [],
        "terminate": [],
        "sources": [],
        # with validFrom, validTo, citation juriconnect and text
        "explanation": ""
    }
    return act_frame

def create_empty_fact_frame() -> dict:
    fact_frame = {
        "id":"",
        "fact": "",
        "function": [],
        "sources": [],  # with validFrom, validTo, citation juriconnect and text
        "explanation": ""
    }
    return fact_frame

def create_empty_duty_frame() -> dict:
    duty_frame = {
        "id":"",
        "duty": "",
        "duty holder": "",
        "claimant": "",
        "terminating act": [],
        "creating act": [],
        "enforcing act": "",
        "sources": [],
    }
    return duty_frame
